src/dvclive/system_metrics.py

```diff
@@ -0,0 +1,87 @@
+import logging
+from typing import Dict, Union
+
+import psutil
+from statistics import mean
+from threading import Event, Thread
+
+from .utils import append_dict
+
+logger = logging.getLogger("dvclive")
+MEGABYTES_DIVIDER = 1024.0**2
+GIGABYTES_DIVIDER = 1024.0**3
+
+MINIMUM_CPU_USAGE_TO_BE_ACTIVE = 30
+
+
+class CPUMetricsCallback:
+    def __init__(
+        self,
+        duration: Union[int, float] = 5,
+        interval: Union[int, float] = 1.0,
+        plot: bool = True,
+    ):
+        self.duration = duration
+        self.interval = interval
```

Comment:
should do at least basic sanity checks for values ?

Reply:
WandB are doing some clipping on the values: https://github.com/wandb/wandb/blob/cdb7ffe67e19402d4a0728de1f1be4471cef702d/wandb/sdk/internal/system/assets/interfaces.py#L120
If we [agree that the user shouldn't have access to the values](https://github.com/iterative/dvclive/pull/773#discussion_r1485707542), then there is no need to sanity check these (but we can check for their validity in unit tests)